path.cabal

```diff
@@ -21,49 +21,49 @@ library
   hs-source-dirs:    src/
   ghc-options:       -Wall -O2
   exposed-modules:   Path, Path.Internal
-  build-depends:     base >= 4 && <5
-                   , exceptions
-                   , filepath < 1.2.0.1 || >= 1.3
-                   , template-haskell
+  build-depends:     aeson
+                   , base       >= 4.7     && < 5.0
                    , deepseq
-                   , aeson
-                   , hashable >= 1.2 && < 1.3
+                   , exceptions >= 0.6     && < 0.9
+                   , filepath   < 1.2.0.1  || >= 1.3
+                   , hashable   >= 1.2     && < 1.3
```

Comment:
> Packages sholud be defined by their Cabal files, not an external entity such as Stackage's resolvers.

I think it's good to differentiate a bit here: The components that are for external consumption, i.e. typically library and executables, should come with the metadata that the consumers need. But testsuites and benchmarks are not for external consumption. Here the metadata should serve the maintainers and Stackage operators well. And those IMO don't benefit from speculative bounds.

But I don't want to blow this issue out of proportion – if you need these bounds for your workflow, please go ahead.

WRT the lower bounds in the library I agree with NorfairKing. I think you might inadvertedly hurt users there and I don't see the problem with simply leaving them off as long as no incompatibilities are known.